README.md

```diff
@@ -118,11 +118,31 @@ Emsdk contains a history of old compiler versions that you can use to maintain
 your migration path. Type `emsdk list --old` to get a list of archived tool and
 SDK versions, and `emsdk install <name_of_tool>` to install it.
 
+### I want to build from source/I want to download a precompiled build!
+
+Some Emsdk Tool and SDK targets refer to packages that are precompiled, and
+no compilation is needed when installing them. Other Emsdk Tools and SDK
+targets come "from source", meaning that they will fetch the source repositories
+using git, and compile the package on demand.
+
+When you run `emsdk list`, it will group the Tools and SDKs under these two
+categories.
+
+To obtain and build latest upstream wasm SDK from source, run
+
+```
+emsdk install sdk-upstream-incoming-64bit
+```
+
+You can use this target for example to bootstrap developing patches to LLVM,
+Binaryen or Emscripten. (After initial installation, use `git remote add`
+in the cloned tree to add your own fork to push changes as patches)
```

Comment:
maybe mention here that this is not necessary - contributors do not need to build llvm and binaryen from source for almost all patches to emscripten. Can link to https://emscripten.org/docs/contributing/developers_guide.html?highlight=developer#setting-up which has more details.